src/cxx/basic_api.cc

```diff
@@ -468,6 +468,79 @@ JNIEXPORT void JNICALL Java_foundation_pEp_jniadapter_Engine__1config_1passphras
     }
 }
 
+JNIEXPORT jobject JNICALL Java_foundation_pEp_jniadapter_Engine__1unlock_1keys_1with_1passphrase(JNIEnv *env,
+         jobject obj,
+         jobject accountswithpassphrases)
+{
+    std::mutex *mutex_local = nullptr;
+    {
+        std::lock_guard<std::mutex> l(global_mutex);
+        pEpLog("called with lock_guard");
+        mutex_local = get_engine_java_object_mutex(env, obj);
+    }
+    std::lock_guard<std::mutex> l(*mutex_local);
+
+    const stringpair_list_t* _accountswithpassphrases = to_stringpairlist(env, accountswithpassphrases);
+    stringlist_t *_errorAccounts;
+
+    PEP_STATUS status = ::unlock_keys_with_passphrase(session(),_accountswithpassphrases,&_errorAccounts);
+
+    jobject errorAccounts_ = NULL;
+    if (status != PEP_STATUS_OK) {
+        if (_errorAccounts) {
+            errorAccounts_ = from_stringlist(env, _errorAccounts);
+        } else {
+            throw_pEp_Exception(env, status);
+            return NULL;
+        }
+    } else {
+        for (const ::stringpair_list_t *curr = _accountswithpassphrases; curr != nullptr; curr = curr->next) {
+            char* passphrase = curr->value->value;
+            passphrase_cache.add(passphrase);
+        }
+    }
+    free_stringlist(_errorAccounts);
+    return errorAccounts_;
+}
+
+JNIEXPORT jobject JNICALL Java_foundation_pEp_jniadapter_Engine__1manage_1passphrase(JNIEnv *env,
+         jobject obj,
+         jobject accountswitholdpassphrases,
+         jbyteArray newpassphrase)
+{
+    std::mutex *mutex_local = nullptr;
+    {
+        std::lock_guard<std::mutex> l(global_mutex);
+        pEpLog("called with lock_guard");
+        mutex_local = get_engine_java_object_mutex(env, obj);
+    }
+    std::lock_guard<std::mutex> l(*mutex_local);
+
+    stringpair_list_t* _accountswitholdpassphrases = to_stringpairlist(env, accountswitholdpassphrases);
+    const char *_newpassphrase = to_string(env, newpassphrase);
+    stringlist_t *_errorAccounts;
```

Comment:
Initialize to nullptr, don't rely on core.

Reply:
Done.